### PR TITLE
Bug 560445 - [YARS] support incremental monitoring

### PR DESCRIPTION
**Pull request by eparovyshnaya**
- design yars incremental monitoring interface
 - implement `Product customers export` in a monitorable and cancellable
way

Signed-off-by: elena.parovyshnaya <elena.parovyshnaya@gmail.com>

### CODE REVIEW

File: bundles/org.eclipse.passage.loc.yars.api/src/org/eclipse/passage/loc/yars/internal/api/Progress.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+	void report(String info);
+
+	boolean cancelDemanded();
+
+	/**
+	 * Empty implementation of the {@linkplain Progress} interfaces, no action is
+	 * taken on any event.
+	 */
+	class Inane<T> implements Progress<T> {
```
Comment (ruspl-afed) [inline bundles/org.eclipse.passage.loc.yars.api/src/org/eclipse/passage/loc/yars/internal/api/Progress.java:38]: final?

File: bundles/org.eclipse.passage.loc.yars.api/src/org/eclipse/passage/loc/yars/internal/api/ExportData.java [hunk 2]
```diff
@@ -58,9 +58,30 @@
...
-	 * @param M any sub type of {@linkplain ListMedia}
+	 * @param media    any sub type of {@linkplain ListMedia}
+	 * @param progress implementation of {@linkplain Progress} interface to report
+	 *                 the state of the export process
 	 * @since 0.1
 	 */
-	void write(M media);
+	void write(M media, Progress<T> progress);
+
+	class Smart<T, M extends ListMedia<T>> implements ExportData<T, M> {
```
Comment (ruspl-afed) [inline bundles/org.eclipse.passage.loc.yars.api/src/org/eclipse/passage/loc/yars/internal/api/ExportData.java:68]: final?

File: bundles/org.eclipse.passage.loc.report.ui/src/org/eclipse/passage/loc/report/internal/ui/jface/VisibleProgress.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
+package org.eclipse.passage.loc.report.internal.ui.jface;
```
Comment (ruspl-afed) [inline bundles/org.eclipse.passage.loc.report.ui/src/org/eclipse/passage/loc/report/internal/ui/jface/VisibleProgress.java:1]: license header

File: bundles/org.eclipse.passage.loc.report.ui/src/org/eclipse/passage/loc/report/internal/ui/jface/MonitoredExportOperation.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
+package org.eclipse.passage.loc.report.internal.ui.jface;
```
Comment (ruspl-afed) [inline bundles/org.eclipse.passage.loc.report.ui/src/org/eclipse/passage/loc/report/internal/ui/jface/MonitoredExportOperation.java:1]: license header

File: bundles/org.eclipse.passage.loc.report.ui/src/org/eclipse/passage/loc/report/internal/ui/jface/VisibleProgress.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+	@Override
+	public void reportNodeSrart(ProductCustomer entry) {
+		monitor.subTask(entry.toString());
+	}
+
+	@Override
+	public void reportNodeFinish(ProductCustomer entry) {
+		monitor.worked(1);
+
```
Comment (ruspl-afed) [inline bundles/org.eclipse.passage.loc.report.ui/src/org/eclipse/passage/loc/report/internal/ui/jface/VisibleProgress.java:31]: empty line

File: bundles/org.eclipse.passage.loc.report.ui/META-INF/MANIFEST.MF [hunk 6]
```diff
@@ -13,6 +13,7 @@ Require-Bundle: org.eclipse.passage.loc.yars.api;bundle-version="0.1.0";visibili
  org.eclipse.jface,
  org.eclipse.osgi.services;bundle-version="3.8.0",
  org.eclipse.passage.lic.products;bundle-version="0.5.200",
- org.eclipse.osgi
+ org.eclipse.osgi,
+ org.eclipse.core.runtime
```
Comment (ruspl-afed) [inline bundles/org.eclipse.passage.loc.report.ui/META-INF/MANIFEST.MF:17]: `org.eclipse.equinox.common` to require less?

File: bundles/org.eclipse.passage.loc.yars.api/src/org/eclipse/passage/loc/yars/internal/api/Progress.java (hunk 1 above)
Comment (eparovyshnaya) [reply bundles/org.eclipse.passage.loc.yars.api/src/org/eclipse/passage/loc/yars/internal/api/Progress.java]: Done

File: bundles/org.eclipse.passage.loc.yars.api/src/org/eclipse/passage/loc/yars/internal/api/ExportData.java (hunk 2 above)
Comment (eparovyshnaya) [reply bundles/org.eclipse.passage.loc.yars.api/src/org/eclipse/passage/loc/yars/internal/api/ExportData.java]: Done

File: bundles/org.eclipse.passage.loc.report.ui/src/org/eclipse/passage/loc/report/internal/ui/jface/VisibleProgress.java (hunk 3 above)
Comment (eparovyshnaya) [reply bundles/org.eclipse.passage.loc.report.ui/src/org/eclipse/passage/loc/report/internal/ui/jface/VisibleProgress.java:13]: Done

File: bundles/org.eclipse.passage.loc.report.ui/src/org/eclipse/passage/loc/report/internal/ui/jface/MonitoredExportOperation.java (hunk 4 above)
Comment (eparovyshnaya) [reply bundles/org.eclipse.passage.loc.report.ui/src/org/eclipse/passage/loc/report/internal/ui/jface/MonitoredExportOperation.java:13]: Done

File: bundles/org.eclipse.passage.loc.report.ui/src/org/eclipse/passage/loc/report/internal/ui/jface/VisibleProgress.java (hunk 5 above)
Comment (eparovyshnaya) [reply bundles/org.eclipse.passage.loc.report.ui/src/org/eclipse/passage/loc/report/internal/ui/jface/VisibleProgress.java]: Removed

File: bundles/org.eclipse.passage.loc.report.ui/META-INF/MANIFEST.MF (hunk 6 above)
Comment (eparovyshnaya) [reply bundles/org.eclipse.passage.loc.report.ui/META-INF/MANIFEST.MF:17]: Have not got the idea, can you please explain a bit wider? `IProgressMonitor` is declared in `core.runtime`. How I can avoid depending on it?

Comment (ruspl-afed) [reply bundles/org.eclipse.passage.loc.report.ui/META-INF/MANIFEST.MF:17]: It is declared in this package, but the bundle is `org.eclipse.equinox.common`

Comment (ruspl-afed) [reply bundles/org.eclipse.passage.loc.report.ui/META-INF/MANIFEST.MF:17]: can wait until special work to align our requirements with particular platform release. The idea is to keep LIC as low as possible while use latest improvements for LBC, LOC, LDC.

**Pull request merged**